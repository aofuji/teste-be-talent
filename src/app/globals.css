@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --blue-primary: #0500ff;
  --blue-10: #edeffb;
  --black: #1c1c1c;
  --gray-20: #9e9e9e;
  --gray-10: #dfdfdf;
  --gray-05: #f5f5f5;
  --gray-00: #f0f0f0;
  --white: #ffffff;
}

body {
  color: var(--foreground);
  background: var(--background);
}

h1 {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: medium;
  font-size: 21px;
}

h2 {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: medium;
  font-size: 16px;
}

h3 {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: medium;
  font-size: 14px;
}
