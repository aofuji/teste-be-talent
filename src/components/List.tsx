"use client";

import { Employee } from "@/interfaces/employees";
import formatDate from "@/utils/formatDate";
import formatPhone from "@/utils/formatPhone";
import Image from "next/image";

const List = ({ data }: { data: Array<Employee>}) => {
  return (
    <div className="pr-5 pl-5">
      <table className="min-w-full">
        <thead>
          <tr className="bg-blue_primary text-white uppercase">
            <th className="pl-6 p-2 text-left rounded-tl-xl">foto</th>
            <th className="p-2 text-left">nome</th>
            <th className="p-2 text-left">cargo</th>
            <th className="p-2 text-left">data de admissão</th>
            <th className="p-2 text-left rounded-tr-xl">telefone</th>
          </tr>
        </thead>
        <tbody className="border bg-white">
          { data.length === 0 ? (
            <tr>
              <td colSpan={5} className="text-center p-2">Nenhum funcionário encontrado.</td>
            </tr>
          ) : (
            data.map((employee, index) => (
              <tr key={index} className="border-b border-gray_10">
                <td className="pl-6 p-2">
                  <Image
                    src={employee.urlImg}
                    alt={employee.name}
                    width={40}
                    height={40}
                    className="rounded-full h-10 object-cover"
                  />
                </td>
                <td className="p-2">{employee.name}</td>
                <td className="p-2">{employee.position}</td>
                <td className="p-2">{formatDate(employee.admissionDate)}</td>
                <td className="p-2">{formatPhone(employee.phone)}</td>
              </tr>
            ))
          )}
        </tbody>
      </table>
    </div>
  );
};

export default List;