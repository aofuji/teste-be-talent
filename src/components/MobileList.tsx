"use client";

import { useState } from "react";
import { Employee } from "@/interfaces/employees";
import Image from "next/image";
import React from "react";
import formatDate from "@/utils/formatDate";
import formatPhone from "@/utils/formatPhone";

const MobileList = ({ data }: { data: Array<Employee> }) => {
  const [expandedRows, setExpandedRows] = useState<{ [key: number]: boolean }>({});

  const handleRowClick = (index: number) => {
    setExpandedRows((prev) => ({
      ...prev,
      [index]: !prev[index],
    }));
  };

  return (
    <div className="pr-5 pl-5">
      <table className="w-full">
        <thead>
          <tr className="bg-blue-700 text-white uppercase">
            <th className="pl-6 p-2 text-left rounded-tl-xl">foto</th>
            <th className="p-2 text-left">nome</th>
            <th className="p-2 text-center rounded-tr-xl">
              <span className="inline-block h-2 w-2 bg-white rounded-full"></span>
            </th>
          </tr>
        </thead>
        <tbody className="border bg-white">
          {data.length === 0 ? (
            <tr>
              <td colSpan={3} className="text-center p-2">
                Nenhum funcionário encontrado.
              </td>
            </tr>
          ) : (
            data.map((employee, index) => (
              <React.Fragment key={index}>
                <tr
                  className={`border-b border-gray-300 cursor-pointer ${expandedRows[index] ? 'border-b-0' : ''}`}
                  onClick={() => handleRowClick(index)}
                >
                  <td className="pl-6 p-2">
                    <Image
                      src={employee.urlImg}
                      alt={employee.name}
                      width={40}
                      height={40}
                      className="rounded-full h-10"
                    />
                  </td>
                  <td className="p-2">{employee.name}</td>
                  <td className="p-2 flex justify-center items-center">
                    <Image
                      src={expandedRows[index] ? "/img/chevron-up.png" : "/img/chevron-down.png"}
                      alt="Chevron"
                      width={32}
                      height={32}
                    />
                  </td>
                </tr>
                {expandedRows[index] && (
                  <tr className="border-b border-gray-300">
                    <td colSpan={3} className="p-2">
                      <div className="p-2 bg-white">
                        <p className="flex justify-between border-b-2 border-dotted border-gray-100 "><strong>Cargo</strong> {employee.position}</p>
                        <p className="flex justify-between border-b-2 border-dotted border-gray-100"><strong>Data de Admissão</strong> {formatDate(employee.admissionDate)}</p>
                        <p className="flex justify-between border-b-2 border-dotted border-gray-100"><strong>Telefone</strong> {formatPhone(employee.phone)}</p>
                      </div>
                    </td>
                  </tr>
                )}
              </React.Fragment>
            ))
          )}
        </tbody>
      </table>
    </div>
  );
};

export default MobileList;