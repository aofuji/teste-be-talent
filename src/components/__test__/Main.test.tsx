import { render } from "@testing-library/react";

import { expect, it } from "@jest/globals";
import List from "../List";
import MobileList from "../MobileList";

it("renders component List", () => {
  const { container } = render(<List data={[]} />);
  expect(container).toMatchSnapshot();
});

it("renders component Mobile List", () => {
  const { container } = render(<MobileList data={[]} />);
  expect(container).toMatchSnapshot();
});
