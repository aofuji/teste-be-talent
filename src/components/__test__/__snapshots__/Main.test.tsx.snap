// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component List 1`] = `
<div>
  <div
    class="pr-5 pl-5"
  >
    <table
      class="min-w-full"
    >
      <thead>
        <tr
          class="bg-blue_primary text-white uppercase"
        >
          <th
            class="pl-6 p-2 text-left rounded-tl-xl"
          >
            foto
          </th>
          <th
            class="p-2 text-left"
          >
            nome
          </th>
          <th
            class="p-2 text-left"
          >
            cargo
          </th>
          <th
            class="p-2 text-left"
          >
            data de admissão
          </th>
          <th
            class="p-2 text-left rounded-tr-xl"
          >
            telefone
          </th>
        </tr>
      </thead>
      <tbody
        class="border bg-white"
      >
        <tr>
          <td
            class="text-center p-2"
            colspan="5"
          >
            Nenhum funcionário encontrado.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`renders component Mobile List 1`] = `
<div>
  <div
    class="pr-5 pl-5"
  >
    <table
      class="w-full"
    >
      <thead>
        <tr
          class="bg-blue_primary text-white uppercase"
        >
          <th
            class="pl-6 p-2 text-left rounded-tl-xl"
          >
            foto
          </th>
          <th
            class="p-2 text-left"
          >
            nome
          </th>
          <th
            class="p-2 text-center rounded-tr-xl"
          >
            <span
              class="inline-block h-2 w-2 bg-white rounded-full"
            />
          </th>
        </tr>
      </thead>
      <tbody
        class="border bg-white"
      >
        <tr>
          <td
            class="text-center p-2"
            colspan="3"
          >
            Nenhum funcionário encontrado.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
